File: frontend/src/types/employee.ts
export interface Employee {
  id: string;
  name: string;
  email: string;
  department?: string;
  position?: string;
  created_at: string;
  updated_at: string;
}

export interface EmployeeCreate {
  name: string;
  email: string;
  department?: string;
  position?: string;
}

export interface EmployeeUpdate {
  name?: string;
  email?: string;
  department?: string;
  position?: string;
}

File: backend/app/models/employee.py
from pydantic import BaseModel, EmailStr
from typing import Optional
from uuid import UUID


class EmployeeBase(BaseModel):
    name: str
    email: EmailStr
    department: Optional[str] = None
    position: Optional[str] = None


class EmployeeCreate(EmployeeBase):
    pass


class EmployeeUpdate(BaseModel):
    name: Optional[str] = None
    email: Optional[EmailStr] = None
    department: Optional[str] = None
    position: Optional[str] = None


class Employee(EmployeeBase):
    id: UUID
    tenant_id: UUID
    created_at: str
    updated_at: str
    
    class Config:
        from_attributes = True

